src/Symfony/Component/Messenger/Transport/NullTransport.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+
+/**
+ * Transport that send nothing. Useful for testing purpose.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class NullTransport implements TransportInterface
+{
+    /**
+     * @var Envelope[]
+     */
+    private $sent = array();
+
+    /**
+     * Is the transport stopped?
+     *
+     * @var bool
+     */
+    private $stopped = false;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        if (!$this->stopped) {
```

Comment:
It would be better with an early return.

```
if ($this->stopped) {
    return;
}
```

Reply:
Fixed all comments below